scenes/game_elements/props/eternal_loom/components/eternal_loom.gd

```diff
@@ -48,6 +49,9 @@ func _on_interaction_ended() -> void:
 func on_offering_succeeded() -> void:
 	loom_offering_animation_player.play(&"loom_offering")
 	await loom_offering_animation_player.animation_finished
+	#Mark mission completed
+	var quest_path: String = GameState._state.get_value(GameState.QUEST_SECTION, GameState.QUEST_PATH_KEY, "")
+	GameState.add_completed_mission(quest_path)
```

Comment:
The `_` prefix on the `_state` variable is a convention that means that the variable is private to the class that defines it. https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#functions-and-variables

In this case `_state` is private to `game_state.gd`, so accessing it directly from `eternal_loom.gd` is not good style.

I think you can simplify this API. You don't actually need the quest path here, so you can change this to:

```suggestion
	GameState.mark_quest_completed()
```

and adjust `game_state.gd` as follows.